Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -478,16 +483,17 @@ def main():
             demisto.incidents(incidents)
             demisto.setLastRun(new_last_run)
         elif command == "fetch-events":
-            for events, offset, total_events_count in fetch_events_command(  # noqa: B007
+            for events, offset, total_events_count, new_from_time in fetch_events_command(  # noqa: B007
                 client,
                 params.get("fetchTime"),
-                params.get("fetchLimit"),
+                int(params.get("fetchLimit", 20)),
                 params.get("configIds"),
                 ctx=get_integration_context() or {},
             ):
-                send_events_to_xsiam(events, VENDOR, PRODUCT, should_update_health_module=False)
-                set_integration_context({"offset": offset})
-            demisto.updateModuleHealth({'eventsPulled': total_events_count})
+                if events:
+                    send_events_to_xsiam(events, VENDOR, PRODUCT, should_update_health_module=False)
+                set_integration_context({"offset": offset, "last_run_time": new_from_time})
+            demisto.updateModuleHealth({'eventsPulled': (total_events_count or 0)})
```

Comment:
why is the "or 0" necessary?

Reply:
to cover cases where total_events_count is not defined (happened according to the logs).